### PR TITLE
AI and silicon code refactor. Multicam and AI eye port from /tg/

### PR DESCRIPTION
**Pull request by DominikPanic**
Yep, exactly how it sounds like.

:cl: LaKiller8
refactor: Silicon code has been refactored.
code: AI eye and multicam code has been ported from /tg/.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Merge conflict.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -451,166 +194,201 @@ var/list/ai_verbs_default = list(
...
+					if("chicken")
+						holo_icon = getHologramIcon(icon(icon_list[input], "chicken_brown"))
+					if("spider")
+						holo_icon = getHologramIcon(icon(icon_list[input], "guard"))
+					else
+						holo_icon = getHologramIcon(icon(icon_list[input], input))
 		else
-			alert("No suitable records found. Aborting.")
+			var/list/icon_list = list(
+				"default" = 'icons/mob/ai.dmi',
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/modules/mob/living/silicon/ai/ai.dm]: Yeah I copypasted that one from /tg/ along with the code for this.

File: code/game/machinery/computer/camera_console.dm [hunk 2]
```diff
@@ -182,16 +212,17 @@
 /obj/machinery/computer/security/almayer
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/game/machinery/computer/camera_console.dm]: It's either this or even more map changes, though I can do it.

File: code/game/machinery/camera/tracking.dm [hunk 3]
```diff
@@ -1,244 +1,160 @@
...
 		return
-	user.eyeobj.setLoc(get_turf(src))
-
+	user.switchCamera(src)
 
-/mob/living/silicon/ai/attack_ai(var/mob/user as mob)
-	ai_camera_list()
 
-/proc/camera_sort(list/L) // TODO: replace this bubblesort with a mergesort - spookydonut
+/proc/camera_sort(list/L)
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/game/machinery/camera/tracking.dm]: Copypasting the file removed the comment, whoops.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 4]
```diff
@@ -451,166 +194,201 @@ var/list/ai_verbs_default = list(
...
+			lit_cameras = list()
+		to_chat(src, "<span class='notice'>Camera lights deactivated.</span>")
 	else
-		lightNearbyCamera()
+		light_cameras()
+		to_chat(src, "<span class='notice'>Camera lights activated.</span>")
+	camera_light_on = !camera_light_on
 
 
+/mob/living/silicon/ai/proc/light_cameras()
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/modules/mob/living/silicon/ai/ai.dm]: Agreed, though currently only AI can trigger it and AI will be admin only at first.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 5]
```diff
@@ -451,166 +194,201 @@ var/list/ai_verbs_default = list(
...
-		if(personnel_list.len)
 			input = input("Select a crew member:") as null|anything in personnel_list
 			var/icon/character_icon = personnel_list[input]
 			if(character_icon)
 				qdel(holo_icon)//Clear old icon so we're not storing it in memory.
 				holo_icon = getHologramIcon(icon(character_icon))
+
+		if("Animal")
+			var/list/icon_list = list(
+			"bear" = 'icons/mob/animal.dmi',
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/modules/mob/living/silicon/ai/ai.dm]: Yep, copypasting this one too.

File: code/datums/mob_hud.dm [hunk 6]
```diff
@@ -1,4 +1,5 @@
 /* HUD DATUMS */
+GLOBAL_LIST_EMPTY(all_huds)
 
 //GLOBAL HUD LIST
 var/datum/mob_hud/huds = list(
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/datums/mob_hud.dm]: I can do it but I would have to replace it in multiple files thus making this gigantic PR even larger.

File: code/game/machinery/computer/camera_console.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (spookydonut) [reply code/game/machinery/computer/camera_console.dm]: straight type path changes are fine

File: code/datums/mob_hud.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (spookydonut) [reply code/datums/mob_hud.dm]: true

**Pull request merged**